mycore-base/src/main/java/org/mycore/datamodel/metadata/MCRMetadataManager.java

```diff
@@ -296,16 +306,26 @@ private static void importDerivate(String derivateID, Path sourceDir) throws MCR
      * @throws MCRAccessException if "create-{objectType}" privilege is missing
      */
     public static void create(final MCRObject mcrObject) throws MCRPersistenceException, MCRAccessException {
-        String createBasePrivilege = "create-" + mcrObject.getId().getBase();
-        String createTypePrivilege = "create-" + mcrObject.getId().getTypeId();
+
+        MCRObjectID objectId = mcrObject.getId();
+
+        String createBasePrivilege = "create-" + objectId.getBase();
+        String createTypePrivilege = "create-" + objectId.getTypeId();
         if (!MCRAccessManager.checkPermission(createBasePrivilege)
             && !MCRAccessManager.checkPermission(createTypePrivilege)) {
-            throw MCRAccessException.missingPrivilege("Create object with id " + mcrObject.getId(), createBasePrivilege,
+            throw MCRAccessException.missingPrivilege("Create object with id " + objectId, createBasePrivilege,
                 createTypePrivilege);
         }
         // exist the object?
-        if (MCRMetadataManager.exists(mcrObject.getId())) {
-            throw new MCRPersistenceException("The object " + mcrObject.getId() + " allready exists, nothing done.");
+        if (MCRMetadataManager.exists(objectId)) {
+            throw new MCRPersistenceException("The object " + objectId + " allready exists, nothing done.");
+        }
+
+        // assign new id if necessary
+        if (objectId.getNumberAsInteger() == 0) {
+            objectId = MCRObjectID.getNextFreeId(objectId.getBase(), objectId.getTypeId());
```

Comment:
I think this has to be .getNextFreeId(objectId.getProjectId(), ...

Also the parameter name of getNextFreeId is wrong. It should be projectIid